src/main/java/org/apache/sling/discovery/base/connectors/ping/TopologyConnectorServlet.java

```diff
@@ -206,39 +203,37 @@ protected void doPut(HttpServletRequest request, HttpServletResponse response)
         }
 
         final String[] pathInfo = request.getPathInfo().split("\\.");
-        final String extension = pathInfo.length==3 ? pathInfo[2] : "";
+        final String extension = pathInfo.length == 3 ? pathInfo[2] : "";
         if (!"json".equals(extension)) {
             response.sendError(HttpServletResponse.SC_NOT_FOUND);
             return;
         }
-        
-        final String selector = pathInfo.length==3 ? pathInfo[1] : "";
+
+        final String selector = pathInfo.length == 3 ? pathInfo[1] : "";
 
         String topologyAnnouncementJSON = requestValidator.decodeMessage(request);
-    	if (logger.isDebugEnabled()) {
-	        logger.debug("doPost: incoming topology announcement is: "
-	                + topologyAnnouncementJSON);
-    	}
+
+        // javasecurity:S5145: Replace pattern-breaking characters
+        logger.debug("doPost: incoming topology announcement is: " + topologyAnnouncementJSON.replaceAll("[\n\r\t]", "_"));
```

Comment:
very minor, but what about keeping the `isDebugEnabled()` guard?

```suggestion
        if (logger.isDebugEnabled()) {
            // javasecurity:S5145: Replace pattern-breaking characters
            logger.debug("doPost: incoming topology announcement is: " + topologyAnnouncementJSON.replaceAll("[\n\r\t]", "_"));
        }
```

Reply:
Fixed ✅